src/cmds/openapi/reduce.ts

```diff
@@ -0,0 +1,189 @@
+    ];
+  }
+
+  async run(opts: CommandOptions<Options>) {
+    super.run(opts);
+
+    const { spec, workingDirectory } = opts;
+
+    if (workingDirectory) {
+      process.chdir(workingDirectory);
+    }
+
+    const { bundledSpec, specPath, specType } = await prepareOas(spec, 'openapi:reduce');
+    const parsedBundledSpec = JSON.parse(bundledSpec);
+
+    if (specType !== 'OpenAPI') {
+      throw new Error('Sorry, this reducer feature in rdme only supports OpenAPI 3.0+ definitions.');
+    }
+
+    const promptResults = await promptTerminal([
+      {
+        type: 'select',
+        name: 'reduceBy',
+        message: 'Would you like to reduce by paths or tags?',
+        choices: [
+          { title: 'Tags', value: 'tags' },
+          { title: 'Paths', value: 'paths' },
+        ],
+      },
+      {
+        type: (prev, values) => (values.reduceBy === 'tags' ? 'multiselect' : null),
+        name: 'tags',
+        message: 'Choose which tags to reduce by:',
+        min: 1,
+        choices: () => {
+          const tags = jsonpath
+            .query(parsedBundledSpec, '$..paths..tags')
+            .flat()
+            .filter(tag => {
+              // A spec may have a `name: tag` which this JSONPath query will pick up. Since that
+              // definitely isn't a tag we want here we need to filter them out.
+              return typeof tag === 'string';
+            });
+
+          return [...new Set(tags)].map(tag => ({
+            title: tag,
+            value: tag,
+          }));
+        },
+      },
+      {
+        type: (prev, values) => (values.reduceBy === 'paths' ? 'multiselect' : null),
+        name: 'paths',
+        message: 'Choose which paths to reduce by:',
+        min: 1,
+        choices: () => {
+          return Object.keys(parsedBundledSpec.paths).map(p => ({
+            title: p,
+            value: p,
+          }));
+        },
+      },
+      {
+        type: (prev, values) => (values.reduceBy === 'paths' ? 'multiselect' : null),
+        name: 'methods',
+        message: 'Choose which HTTP methods that are available across these paths to reduce by:',
+        min: 1,
+        choices: (prev, values) => {
+          const paths: string[] = values.paths;
+          let methods = paths
+            .map((p: string) => Object.keys(parsedBundledSpec.paths[p]))
+            .flat()
+            .filter((method: string) => method.toLowerCase() !== 'parameters');
+
+          methods = [...new Set(methods)];
+          methods.sort();
+
+          return methods.map((method: string) => ({
+            title: method.toUpperCase(),
+            value: method,
+          }));
+        },
+      },
+      {
+        type: 'text',
+        name: 'outputPath',
+        message: 'Enter the path to save your reduced API definition to:',
+        initial: () => {
+          const extension = path.extname(specPath);
+          return `${path.basename(specPath).split(extension)[0]}-reduced${extension}`;
```

Comment:
This'll dynamically turn `petstore.json` or `petstore.yaml` into `petstore-reduced.{ext}`

Reply:
:nice-rgb: